PyRDF/RDataFrame.py

```diff
@@ -8,11 +9,58 @@ class RDataFrame(Node):
     of ROOT C++'s RDataFrame 
     data structure
 
+    Supported constructor arguments :
+    - Number of entries (Int)
+        * For eg., RDataFrame(64)
+    - treename (str), fileslist (str or list of str)
+        * For eg., RDataFrame("myTree", "file.root") or RDataFrame("myTree", ["file1.root", "file2.root"])
+    - treename (str), fileslist (str or list of str), default branches (str or list of str)
+        * For eg., RDataFrame("myTree", ["file1.root", "file2.root"], ["branch1", "branch2"])
+
     """
 
-    def __init__(self, treename, filelist):
-        # TODO (shravan97) : Implement different types of constructors
+    def __init__(self, *args):
 
         super(RDataFrame, self).__init__(None, None)
-        self.filelist = filelist
-        self.treename = treename
\ No newline at end of file
+
+        args = list(args) # Make args mutable
+        num_params = len(args)
+
+        for i in range(num_params):
+            if isinstance(args[i], list):
+                args[i] = self._get_vector_from_list(args[i])
+
+        try:
+            ROOT.ROOT.RDataFrame(*args) # Check if the args are correct
+        except TypeError as e:
+            print("Error creating the RDataFrame !")
+            raise RDataFrameException(e) from None
```

Comment:
Can you perhaps construct the `RDataFrameException` with two arguments: the previous exception and a message that you will prepend? This way you can eliminate the `print` from here and do everything inside `RDataFrameException`